cdptools/databases/document_store_database.py

```diff
@@ -0,0 +1,840 @@
+    gt: str = ">"
+    lt: str = "<"
+    gteq: str = ">="
+    lteq: str = "<="
+
+
+class OrderOperators:
+    desc: str = "DESCENDING"
+    asce: str = "ASCENDING"
+
+
+class TermResult(NamedTuple):
+    term: str
+    contribution: float
+
+
+class Match:
+    def __init__(self, unique_id, terms: List[TermResult], data: Dict[str, Any]):
+        self._unique_id = unique_id
+        self._terms = terms
+        self._data = data
+
+    @property
+    def unique_id(self):
+        return self._unique_id
+
+    @property
+    def data(self):
+        return self._data
+
+    @property
+    def terms(self):
+        return self._terms
+
+    @property
+    def relevance(self):
+        return sum(t.contribution for t in self.terms)
+
+    def __str__(self):
+        return f"<Match [unique_id: {self.unique_id}, relevance: {self.relevance}]>"
+
+    def __repr__(self):
+        return str(self)
+
+
+###############################################################################
+
+
+class DocumentStoreDatabase(ABC):
+    @staticmethod
+    def _construct_where_condition(filt: Union[WhereCondition, List, Tuple]):
+        """
+        Construct a where condition from the passed filter object.
+
+        For available where condition operators, please view the WhereOperators class 
+        attributes.
+
+        Parameters
+        ----------
+        filt: Union[WhereCondition, List, Tuple]
+            The filter to construct a WhereCondition for.
+
+        Returns
+        -------
+        filt: WhereCondition
+            If provided an already constructed WhereCondition, simply returns the 
+            provided object.
+            If provided a List or Tuple of length two, constructs a WhereCondition 
+            with an equal operator.
+            If provided a List or Tuple of length three, constructs a WhereCondition 
+            with the contents of the provided.
+            Provided List or Tuple objects should follow the format:
+                {target_to_filter}, {operator}, {target_value}
+            Unless provided a List or Tuple of length two, in which case, should 
+            follow the format:
+                {target_to_filter}, {target_value}
+
+        Examples
+        --------
+        ```
+        condition = Database._construct_where_condition(WhereCondition(
+            column_name="event_id",
+            operator=">=",
+            value="abcd"
+        ))
+        # Returns the same WhereCondition that was passed in
+
+        condition = Database._construct_where_condition(["event_id", 
+                                                        WhereOperators.gteq, 
+                                                        value="abcd"])
+        # Returns the same WhereCondition as above
+
+        condition = Database._construct_where_condition(("event_id", "abcd"))
+        # Returns a WhereCondition similar to above but the operator is set to equal 
+        rather than greater than or equal.
```

Comment:
I found out that the `numpy` examples docstring is not like this. This is my very much my bad but can you update these: https://numpydoc.readthedocs.io/en/latest/format.html#id15 go to `15. Example` section for an... _example_ of an _example_ docstring. :upside_down_face:

Reply:
If I'm understanding the comment correctly I formatted the example docstrings to be in format 
```
>>> INPUT
OUTPUT


# Comment about it
```